### ISSUE TITLE
Header part not reflecting Jupyter Notebook Top Area default setting

### ISSUE THREAD

**Issue by yumyumqing** [jupyter/notebook#6833]
**Describe the bug**
In settings editor, `Jupyter Notebook Top Area` is only reflecting changes from `User Preferences` but not `Settings Overrides`.

When I start the jupyter notebook with default settings overrides, the menu does pick up the `visible: false` setting, but the notebook is still showing header part.
If I toggle the setting manually by clicking the box, the change does reflect in the notebook header part.
But then if I click `Restore to Defaults`, the notebook header is not picking up the change again.

**To Reproduce**

1. Set up dev environment as https://github.com/jupyter/notebook/blob/main/CONTRIBUTING.md#setting-up-a-development-environment
2. Create a `overrides.json` file in `settings` dir in [JupyterLab Application Directory](https://jupyterlab.readthedocs.io/en/stable/user/directories.html#application-directory). 
If using `mamba` as step 1, the dir should look something like `$HOME/mambaforge/envs/notebook/share/jupyter/lab/settings`.
```
{
  "@jupyter-notebook/application-extension:top": {
    "visible": false
  }
}
```
3. Build and launch `jupyter notebook` in dev mode
4. Go to the browser page opened by `jupyter notebook`
5. Click `Settings` - `Settings Editor` - `Jupyter Notebook Top Area`
6. See the box is `not` checked
7. In a new tab, open a notebook in notebook mode, and the `header` is still showing
8. Back to the `setting` page and `check` the box and `uncheck` again
9. Refresh the notebook page, see the `header` is `not` showing now
10. Back to the `setting` page and `check` the box and then click `Restore to Defaults`
11. Refresh the notebook page, see the `header` is `still` showing

**Expected behavior**
In my opinion, I think the `header` part should reflect the default settings (if there is no `User Preference`) for 2 reasons:
1. The user might get confused when a unchecked box can lead to 2 different behaviors
2. For tools like `jupyterlite`, that uses this plugin but does not support `User Preference` override, it would be hard for the users to hide the `header` part by default.
(I think `jupyterlite` uses browser cache for `User Preference` settings and does not allow overrides at build time. Please correct me if I am wrong on this.)

**Screenshots**
<img width="920" alt="Screenshot 2023-04-05 at 8 11 38 PM" src="https://user-images.githubusercontent.com/21135321/230076595-f915afdf-e373-4e03-8eb3-3c9c794582e6.png">
<img width="1117" alt="Screenshot 2023-04-05 at 8 13 02 PM" src="https://user-images.githubusercontent.com/21135321/230076812-290a8e5d-beda-41d4-8d96-9acdca14dd74.png">

**Desktop (please complete the following information):**
- OS: MacOS Ventura 13.2.1
- Browser: chrome
- Version: tip of main branch

**Additional context**
I think the reason for the bug might come from these 2 places:
1. `updateSettings` is only changing top visibility when there is `User Preference` set up. This should probably be outside of the user setting check.
https://github.com/jupyter/notebook/blob/1512c8299284768373c7e40e4c97ceb8d209ae6b/packages/application-extension/src/index.ts#L497-L499
2. Whenever refreshing a page, if the `User Preference` isn't set up, the plugin will adjust the header showing based on screen size only.
https://github.com/jupyter/notebook/blob/1512c8299284768373c7e40e4c97ceb8d209ae6b/packages/application-extension/src/index.ts#L519-L528

If we fix the bug as in **Expected behavior**, this neat feature of adjusting based on screen size might go away.
If we still want to keep the adjustment, maybe we can have another variable to flag whether we would like to do adjustment like `adjustToScreen: true` which could overrides the header settings.

Thanks for taking a look!

**Comment by jtpio** [comment]
Thanks @yumyumqing for opening this issue and providing all the details :+1: 

>  In my opinion, I think the header part should reflect the default settings (if there is no User Preference) for 2 reasons:

Would you like to open a draft PR so we can see how that would work?

> 1. `updateSettings` is only changing top visibility when there is `User Preference` set up.

I don't recall exactly why it is like this. But this git history seems to be pointing to this commit: https://github.com/jupyter/notebook/commit/6c96c80c8c4fa7c02336acf6dd184dc4cba50d0e

**Comment by yumyumqing** [comment]
Hi @jtpio I opened this PR draft and listed a couple different expected behavior cases. https://github.com/jupyter/notebook/pull/6836

Some of the checks seem failing due to the changes though, maybe because the UI changes a little. Can you please help take a look? Thanks a lot!

**Comment by yumyumqing** [comment]
Fixed with PR #6836
